src/main/java/io/jenkins/plugins/oidc_provider/IdTokenCredentials.java

```diff
@@ -181,11 +181,12 @@ RSAPublicKey publicKey() {
     )));
 
     protected final @NonNull String token() {
+        IdTokenConfiguration cfg = IdTokenConfiguration.get();
         JwtBuilder builder = Jwts.builder().
             setHeaderParam("kid", getId()).
             setIssuer(issuer != null ? issuer : findIssuer().url()).
             setAudience(audience).
-            setExpiration(Date.from(new Date().toInstant().plus(1, ChronoUnit.HOURS))).
+            setExpiration(Date.from(new Date().toInstant().plus(cfg.getTokenLifetime(), ChronoUnit.SECONDS))).
```

Comment:
BTW can probably simplify this to
```suggestion
            setExpiration(Date.from(Instant.now().plus(cfg.getTokenLifetime(), ChronoUnit.SECONDS))).
```